plugins/vocabbuilder.koplugin/main.lua

```diff
@@ -0,0 +1,1180 @@
+                        }
+                    },
+                    focus_button
+                },
+                background = Blitbuffer.COLOR_WHITE,
+                bordersize = Size.border.window,
+                radius = Size.radius.window,
+                padding = Size.padding.button,
+                padding_bottom = 0,
+            }
+        }
+    }
+
+end
+
+function WordInfoDialog:setTitle(title)
+    self.title = title
+    self:free()
+    self:init()
+    UIManager:setDirty("all", "ui")
+end
+
+function WordInfoDialog:onShow()
+    UIManager:setDirty(self, function()
+        return "flashui", self[1][1].dimen
+    end)
+end
+
+function WordInfoDialog:onCloseWidget()
+    UIManager:setDirty(nil, function()
+        return "ui", self[1][1].dimen
+    end)
+end
+
+function WordInfoDialog:onTapClose()
+    UIManager:close(self)
+    if self.tap_close_callback then
+        self.tap_close_callback()
+    end
+    return true
+end
+
+function WordInfoDialog:onClose()
+    self:onTapClose()
+    return true
+end
+
+function WordInfoDialog:paintTo(...)
+    InputContainer.paintTo(self, ...)
+    self.dimen = self[1][1].dimen -- FrameContainer
+end
+
+
+
+-- values useful for item cells
+local review_button_width = math.ceil(math.min(Screen:scaleBySize(95), Screen:getWidth()/6))
+local ellipsis_button_width = Screen:scaleBySize(34)
+local star_width = Screen:scaleBySize(25)
+
+local point_widget = TextWidget:new{
+    text = " • ",
+    bold = true,
+    face = Font:getFace("cfont", 24),
+    fgcolor = dim_color
+}
+
+--[[--
+Individual word item widget
+--]]--
+local VocabItemWidget = InputContainer:new{
+    face = Font:getFace("smallinfofont"),
+    width = nil,
+    height = nil,
+    show_parent = nil,
+    item = nil,
+    forgot_button = nil,
+    got_it_button = nil,
+    more_button = nil,
+    layout = nil
+}
+--[[--
+    item: {
+        checked_func: Block,
+        review_count: interger,
+        word: Text
+        book_title: TEXT
+        create_time: Integer
+        review_time: Integer
+        due_time: Integer,
+        got_it_callback: function
+        remove_callback: function
+        is_dim: BOOL
+    }
+--]]--
+
+local point_widget_height = point_widget:getSize().h
+local point_widget_width = point_widget:getSize().w
+local word_height = TextWidget:new{text = " ", face = word_face}:getSize().h
+local subtitle_height = TextWidget:new{text = " ", face = subtitle_face}:getSize().h
+
+
+function VocabItemWidget:init()
+    self.layout = {}
+    self.dimen = Geom:new{w = self.width, h = self.height}
+    self.ges_events.Tap = {
+        GestureRange:new{
+            ges = "tap",
+            range = self.dimen,
+        }
+    }
+    self.ges_events.Hold = {
+        GestureRange:new{
+            ges = "hold",
+            range = self.dimen,
+        }
+    }
+    self.v_spacer = VerticalSpan:new{width = (self.height - word_height - subtitle_height)/2}
```

Comment:
Divided dimensions need to be `math.floor`'ed to avoid Lua shenanigans (everything is a double).

(I noticed a few earlier, too).